dvc/utils/__init__.py

```diff
@@ -88,32 +88,28 @@ def bytes_md5(byts):
     return hasher.hexdigest()
 
 
-def dict_filter(d, exclude=[]):
+def dict_filter(d, exclude=()):
     """
     Exclude specified keys from a nested dict
     """
 
+    def fix_key(k):
+        return str(k) if isinstance(k, builtin_str) else k
+
     if isinstance(d, list):
-        ret = []
-        for e in d:
-            ret.append(dict_filter(e, exclude))
-        return ret
-    elif isinstance(d, dict):
-        ret = {}
-        for k, v in d.items():
-            if isinstance(k, builtin_str):
-                k = str(k)
+        return [dict_filter(e, exclude) for e in d]
```

Comment:
Any point in refactoring this? Again, single-lines are not easy to read. Also, you are breaking git blame by it, right? :slightly_smiling_face:

Reply:
It all started from mutable default, which might be problematic. Many lines are not easy to read) We will change code, this is inevitable.